diff --git a/WORKSPACE b/WORKSPACE
index a049c47..bd2c1e1 100644
--- a/WORKSPACE
+++ b/WORKSPACE
@@ -172,6 +172,7 @@ http_archive(
 new_local_repository(
     name = "linux_opencv",
     build_file = "@//third_party:opencv_linux.BUILD",
+    # update path if necessary
     path = "/usr",
 )
 
diff --git a/third_party/opencv_linux.BUILD b/third_party/opencv_linux.BUILD
index 26978d5..8354d54 100644
--- a/third_party/opencv_linux.BUILD
+++ b/third_party/opencv_linux.BUILD
@@ -15,14 +15,14 @@ cc_library(
     name = "opencv",
     srcs = glob(
         [
-            "lib/x86_64-linux-gnu/libopencv_core.so",
-            "lib/x86_64-linux-gnu/libopencv_calib3d.so",
-            "lib/x86_64-linux-gnu/libopencv_features2d.so",
-            "lib/x86_64-linux-gnu/libopencv_highgui.so",
-            "lib/x86_64-linux-gnu/libopencv_imgcodecs.so",
-            "lib/x86_64-linux-gnu/libopencv_imgproc.so",
-            "lib/x86_64-linux-gnu/libopencv_video.so",
-            "lib/x86_64-linux-gnu/libopencv_videoio.so",
+            "lib/libopencv_core.so",
+            "lib/libopencv_calib3d.so",
+            "lib/libopencv_features2d.so",
+            "lib/libopencv_highgui.so",
+            "lib/libopencv_imgcodecs.so",
+            "lib/libopencv_imgproc.so",
+            "lib/libopencv_video.so",
+            "lib/libopencv_videoio.so",
         ],
     ),
     hdrs = glob([
