diff --git a/mediapipe/calculators/tflite/tflite_inference_calculator.cc b/mediapipe/calculators/tflite/tflite_inference_calculator.cc
index 2c956d6..df8b3c7 100644
--- a/mediapipe/calculators/tflite/tflite_inference_calculator.cc
+++ b/mediapipe/calculators/tflite/tflite_inference_calculator.cc
@@ -431,12 +431,14 @@ mediapipe::Status TfLiteInferenceCalculator::Process(CalculatorContext* cc) {
 
     // 1. Receive pre-processed tensor inputs.
     if (gpu_input_) {
-      MP_RETURN_IF_ERROR(ProcessInputsGpu(cc,
 #if MEDIAPIPE_TFLITE_METAL_INFERENCE
+      MP_RETURN_IF_ERROR(ProcessInputsGpu(cc,
                                           compute_encoder,
-#endif  // MEDIAPIPE_TFLITE_METAL_INFERENCE
-
                                           output_tensors_gpu.get()));
+#else
+      MP_RETURN_IF_ERROR(ProcessInputsGpu(cc,
+                                          output_tensors_gpu.get()));
+#endif  // MEDIAPIPE_TFLITE_METAL_INFERENCE
     } else {
       MP_RETURN_IF_ERROR(ProcessInputsCpu(cc, output_tensors_cpu.get()));
     }
@@ -462,11 +464,13 @@ mediapipe::Status TfLiteInferenceCalculator::Process(CalculatorContext* cc) {
 
     // 3. Output processed tensors.
     if (gpu_output_ || use_advanced_gpu_api_) {
-      MP_RETURN_IF_ERROR(ProcessOutputsGpu(cc, std::move(output_tensors_cpu),
 #if MEDIAPIPE_TFLITE_METAL_INFERENCE
-                                           compute_encoder,
-#endif  // MEDIAPIPE_TFLITE_METAL_INFERENCE
+      MP_RETURN_IF_ERROR(ProcessOutputsGpu(cc, std::move(output_tensors_cpu),
+                                           compute_encoder, std::move(output_tensors_gpu)));
+#else
+      MP_RETURN_IF_ERROR(ProcessOutputsGpu(cc, std::move(output_tensors_cpu),
                                            std::move(output_tensors_gpu)));
+#endif  // MEDIAPIPE_TFLITE_METAL_INFERENCE
     } else {
       MP_RETURN_IF_ERROR(ProcessOutputsCpu(cc, std::move(output_tensors_cpu)));
     }
diff --git a/mediapipe/util/tracking/box_tracker.cc b/mediapipe/util/tracking/box_tracker.cc
index 2111eac..7c5cd2b 100644
--- a/mediapipe/util/tracking/box_tracker.cc
+++ b/mediapipe/util/tracking/box_tracker.cc
@@ -667,7 +667,7 @@ bool BoxTracker::WaitForChunkFile(int id, int checkpoint,
       }
     }
 
-    usleep(wait_time_msec * 1000);
+    absl::SleepFor(absl::Milliseconds(wait_time_msec));
     total_wait_msec += wait_time_msec;
 
     struct stat tmp;
diff --git a/mediapipe/util/tracking/tracking.cc b/mediapipe/util/tracking/tracking.cc
index 72245c0..2a5d93f 100644
--- a/mediapipe/util/tracking/tracking.cc
+++ b/mediapipe/util/tracking/tracking.cc
@@ -1652,7 +1652,7 @@ bool MotionBox::GetVectorsAndWeights(
 
     vectors->push_back(&motion_vectors[k]);
 
-    auto is_close_to_test_vector = [test_vector](const Vector2_f v) -> bool {
+    auto is_close_to_test_vector = [test_vector, kSqProximity](const Vector2_f v) -> bool {
       return (v - test_vector.pos).Norm2() < kSqProximity;
     };
 
