diff --git a/mediapipe/examples/coral/graphs/face_detection_desktop_live.pbtxt b/mediapipe/examples/coral/graphs/face_detection_desktop_live.pbtxt
index 5532128..2067d8f 100644
--- a/mediapipe/examples/coral/graphs/face_detection_desktop_live.pbtxt
+++ b/mediapipe/examples/coral/graphs/face_detection_desktop_live.pbtxt
@@ -69,7 +69,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/examples/coral/models/face-detector-quantized_edgetpu.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face-detector-quantized_edgetpu.tflite"
     }
   }
 }
diff --git a/mediapipe/examples/coral/graphs/object_detection_desktop_live.pbtxt b/mediapipe/examples/coral/graphs/object_detection_desktop_live.pbtxt
index 03bc9e1..8bc9adc 100644
--- a/mediapipe/examples/coral/graphs/object_detection_desktop_live.pbtxt
+++ b/mediapipe/examples/coral/graphs/object_detection_desktop_live.pbtxt
@@ -69,7 +69,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/examples/coral/models/object-detector-quantized_edgetpu.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/object-detector-quantized_edgetpu.tflite"
     }
   }
 }
diff --git a/mediapipe/examples/desktop/autoflip/subgraph/autoflip_object_detection_subgraph.pbtxt b/mediapipe/examples/desktop/autoflip/subgraph/autoflip_object_detection_subgraph.pbtxt
index bd2e7a7..4fc9412 100644
--- a/mediapipe/examples/desktop/autoflip/subgraph/autoflip_object_detection_subgraph.pbtxt
+++ b/mediapipe/examples/desktop/autoflip/subgraph/autoflip_object_detection_subgraph.pbtxt
@@ -37,7 +37,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/examples/desktop/autoflip/subgraph/face_detection_subgraph.pbtxt b/mediapipe/examples/desktop/autoflip/subgraph/face_detection_subgraph.pbtxt
index 2dfb0c5..53916f9 100644
--- a/mediapipe/examples/desktop/autoflip/subgraph/face_detection_subgraph.pbtxt
+++ b/mediapipe/examples/desktop/autoflip/subgraph/face_detection_subgraph.pbtxt
@@ -38,7 +38,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/models/face_detection_back.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_back.tflite"
     }
   }
 }
diff --git a/mediapipe/examples/desktop/autoflip/subgraph/front_face_detection_subgraph.pbtxt b/mediapipe/examples/desktop/autoflip/subgraph/front_face_detection_subgraph.pbtxt
index b88ea0c..57f1914 100644
--- a/mediapipe/examples/desktop/autoflip/subgraph/front_face_detection_subgraph.pbtxt
+++ b/mediapipe/examples/desktop/autoflip/subgraph/front_face_detection_subgraph.pbtxt
@@ -37,7 +37,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/models/face_detection_front.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_front.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/face_detection/face_detection_back_desktop_live.pbtxt b/mediapipe/graphs/face_detection/face_detection_back_desktop_live.pbtxt
index daccc27..52698f0 100644
--- a/mediapipe/graphs/face_detection/face_detection_back_desktop_live.pbtxt
+++ b/mediapipe/graphs/face_detection/face_detection_back_desktop_live.pbtxt
@@ -64,7 +64,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/face_detection_back.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_back.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/face_detection/face_detection_back_mobile_gpu.pbtxt b/mediapipe/graphs/face_detection/face_detection_back_mobile_gpu.pbtxt
index 669b4b9..eb6faf6 100644
--- a/mediapipe/graphs/face_detection/face_detection_back_mobile_gpu.pbtxt
+++ b/mediapipe/graphs/face_detection/face_detection_back_mobile_gpu.pbtxt
@@ -65,7 +65,7 @@ node {
   output_stream: "TENSORS_GPU:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/face_detection_back.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_back.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/face_detection/face_detection_desktop_live.pbtxt b/mediapipe/graphs/face_detection/face_detection_desktop_live.pbtxt
index dd29961..1960759 100644
--- a/mediapipe/graphs/face_detection/face_detection_desktop_live.pbtxt
+++ b/mediapipe/graphs/face_detection/face_detection_desktop_live.pbtxt
@@ -64,7 +64,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/face_detection_front.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_front.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/face_detection/face_detection_mobile_cpu.pbtxt b/mediapipe/graphs/face_detection/face_detection_mobile_cpu.pbtxt
index f3ae28b..2de1356 100644
--- a/mediapipe/graphs/face_detection/face_detection_mobile_cpu.pbtxt
+++ b/mediapipe/graphs/face_detection/face_detection_mobile_cpu.pbtxt
@@ -75,7 +75,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/face_detection_front.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_front.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/face_detection/face_detection_mobile_gpu.pbtxt b/mediapipe/graphs/face_detection/face_detection_mobile_gpu.pbtxt
index 8c79a6c..a6da8c7 100644
--- a/mediapipe/graphs/face_detection/face_detection_mobile_gpu.pbtxt
+++ b/mediapipe/graphs/face_detection/face_detection_mobile_gpu.pbtxt
@@ -65,7 +65,7 @@ node {
   output_stream: "TENSORS_GPU:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/face_detection_front.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_front.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/hair_segmentation/hair_segmentation_desktop_live.pbtxt b/mediapipe/graphs/hair_segmentation/hair_segmentation_desktop_live.pbtxt
index 36c6970..1966db0 100644
--- a/mediapipe/graphs/hair_segmentation/hair_segmentation_desktop_live.pbtxt
+++ b/mediapipe/graphs/hair_segmentation/hair_segmentation_desktop_live.pbtxt
@@ -111,7 +111,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:op_resolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/hair_segmentation.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/hair_segmentation.tflite"
       use_gpu: false
     }
   }
diff --git a/mediapipe/graphs/hair_segmentation/hair_segmentation_mobile_gpu.pbtxt b/mediapipe/graphs/hair_segmentation/hair_segmentation_mobile_gpu.pbtxt
index c8db44d..8e96bae 100644
--- a/mediapipe/graphs/hair_segmentation/hair_segmentation_mobile_gpu.pbtxt
+++ b/mediapipe/graphs/hair_segmentation/hair_segmentation_mobile_gpu.pbtxt
@@ -111,7 +111,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:op_resolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/hair_segmentation.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/hair_segmentation.tflite"
       use_gpu: true
     }
   }
diff --git a/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_cpu.pbtxt b/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_cpu.pbtxt
index 65c7d16..20e8690 100644
--- a/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_cpu.pbtxt
+++ b/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_cpu.pbtxt
@@ -49,7 +49,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:op_resolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/palm_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/palm_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_gpu.pbtxt b/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_gpu.pbtxt
index 8332860..a6433e5 100644
--- a/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_gpu.pbtxt
+++ b/mediapipe/graphs/hand_tracking/subgraphs/hand_detection_gpu.pbtxt
@@ -53,7 +53,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:opresolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/palm_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/palm_detection.tflite"
       use_gpu: true
     }
   }
diff --git a/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_cpu.pbtxt b/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_cpu.pbtxt
index 50881e8..1b4664d 100644
--- a/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_cpu.pbtxt
+++ b/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_cpu.pbtxt
@@ -71,7 +71,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:op_resolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/hand_landmark.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/hand_landmark.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_gpu.pbtxt b/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_gpu.pbtxt
index fa0a00f..cd13778 100644
--- a/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_gpu.pbtxt
+++ b/mediapipe/graphs/hand_tracking/subgraphs/hand_landmark_gpu.pbtxt
@@ -75,7 +75,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:op_resolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/hand_landmark.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/hand_landmark.tflite"
       use_gpu: true
     }
   }
diff --git a/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_cpu.pbtxt b/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_cpu.pbtxt
index 928e752..42572b1 100644
--- a/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_cpu.pbtxt
+++ b/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_cpu.pbtxt
@@ -49,7 +49,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:opresolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/palm_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/palm_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_gpu.pbtxt b/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_gpu.pbtxt
index afd1fd1..463ee53 100644
--- a/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_gpu.pbtxt
+++ b/mediapipe/graphs/hand_tracking/subgraphs/multi_hand_detection_gpu.pbtxt
@@ -53,7 +53,7 @@ node {
   input_side_packet: "CUSTOM_OP_RESOLVER:opresolver"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/palm_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/palm_detection.tflite"
       use_gpu: true
     }
   }
diff --git a/mediapipe/graphs/object_detection/object_detection_desktop_live.pbtxt b/mediapipe/graphs/object_detection/object_detection_desktop_live.pbtxt
index 98b9fab..556819c 100644
--- a/mediapipe/graphs/object_detection/object_detection_desktop_live.pbtxt
+++ b/mediapipe/graphs/object_detection/object_detection_desktop_live.pbtxt
@@ -64,7 +64,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/object_detection/object_detection_desktop_tensorflow_graph.pbtxt b/mediapipe/graphs/object_detection/object_detection_desktop_tensorflow_graph.pbtxt
index f12eeb6..712b370 100644
--- a/mediapipe/graphs/object_detection/object_detection_desktop_tensorflow_graph.pbtxt
+++ b/mediapipe/graphs/object_detection/object_detection_desktop_tensorflow_graph.pbtxt
@@ -27,7 +27,7 @@ node {
   output_side_packet: "SESSION:object_detection_session"
   node_options: {
     [type.googleapis.com/mediapipe.TensorFlowSessionFromSavedModelCalculatorOptions]: {
-      saved_model_path: "mediapipe/models/object_detection_saved_model"
+      saved_model_path: "Assets/Mediapipe/SDK/Models/object_detection_saved_model"
     }
   }
 }
diff --git a/mediapipe/graphs/object_detection/object_detection_desktop_tflite_graph.pbtxt b/mediapipe/graphs/object_detection/object_detection_desktop_tflite_graph.pbtxt
index 15aa2cd..9c36023 100644
--- a/mediapipe/graphs/object_detection/object_detection_desktop_tflite_graph.pbtxt
+++ b/mediapipe/graphs/object_detection/object_detection_desktop_tflite_graph.pbtxt
@@ -56,7 +56,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/object_detection/object_detection_mobile_cpu.pbtxt b/mediapipe/graphs/object_detection/object_detection_mobile_cpu.pbtxt
index 8256179..7f32e82 100644
--- a/mediapipe/graphs/object_detection/object_detection_mobile_cpu.pbtxt
+++ b/mediapipe/graphs/object_detection/object_detection_mobile_cpu.pbtxt
@@ -75,7 +75,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/object_detection/object_detection_mobile_gpu.pbtxt b/mediapipe/graphs/object_detection/object_detection_mobile_gpu.pbtxt
index 1ed66e8..e96645a 100644
--- a/mediapipe/graphs/object_detection/object_detection_mobile_gpu.pbtxt
+++ b/mediapipe/graphs/object_detection/object_detection_mobile_gpu.pbtxt
@@ -65,7 +65,7 @@ node {
   output_stream: "TENSORS_GPU:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/template_matching/index_building.pbtxt b/mediapipe/graphs/template_matching/index_building.pbtxt
index 8228139..54e9a4d 100644
--- a/mediapipe/graphs/template_matching/index_building.pbtxt
+++ b/mediapipe/graphs/template_matching/index_building.pbtxt
@@ -63,7 +63,7 @@ node {
   }
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/knift_float_400.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/knift_float_400.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/template_matching/template_matching_desktop.pbtxt b/mediapipe/graphs/template_matching/template_matching_desktop.pbtxt
index d44a7e5..78a67b9 100644
--- a/mediapipe/graphs/template_matching/template_matching_desktop.pbtxt
+++ b/mediapipe/graphs/template_matching/template_matching_desktop.pbtxt
@@ -52,7 +52,7 @@ node {
   output_stream: "TENSORS:knift_feature_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/knift_float.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/knift_float.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/template_matching/template_matching_mobile_cpu.pbtxt b/mediapipe/graphs/template_matching/template_matching_mobile_cpu.pbtxt
index 2eceacb..07f15ef 100644
--- a/mediapipe/graphs/template_matching/template_matching_mobile_cpu.pbtxt
+++ b/mediapipe/graphs/template_matching/template_matching_mobile_cpu.pbtxt
@@ -62,7 +62,7 @@ node {
   output_stream: "TENSORS:knift_feature_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/knift_float.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/knift_float.tflite"
       delegate { xnnpack {} }
     }
   }
diff --git a/mediapipe/graphs/tracking/subgraphs/object_detection_cpu.pbtxt b/mediapipe/graphs/tracking/subgraphs/object_detection_cpu.pbtxt
index 54d6af3..197db26 100644
--- a/mediapipe/graphs/tracking/subgraphs/object_detection_cpu.pbtxt
+++ b/mediapipe/graphs/tracking/subgraphs/object_detection_cpu.pbtxt
@@ -39,7 +39,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/graphs/tracking/subgraphs/object_detection_gpu.pbtxt b/mediapipe/graphs/tracking/subgraphs/object_detection_gpu.pbtxt
index f3cc2c8..0524fa9 100644
--- a/mediapipe/graphs/tracking/subgraphs/object_detection_gpu.pbtxt
+++ b/mediapipe/graphs/tracking/subgraphs/object_detection_gpu.pbtxt
@@ -39,7 +39,7 @@ node {
   output_stream: "TENSORS_GPU:detection_tensors"
   node_options: {
     [type.googleapis.com/mediapipe.TfLiteInferenceCalculatorOptions] {
-      model_path: "mediapipe/models/ssdlite_object_detection.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/ssdlite_object_detection.tflite"
     }
   }
 }
diff --git a/mediapipe/modules/face_detection/face_detection_front_cpu.pbtxt b/mediapipe/modules/face_detection/face_detection_front_cpu.pbtxt
index fda86fc..a292eea 100644
--- a/mediapipe/modules/face_detection/face_detection_front_cpu.pbtxt
+++ b/mediapipe/modules/face_detection/face_detection_front_cpu.pbtxt
@@ -59,7 +59,7 @@ node {
   output_stream: "TENSORS:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/modules/face_detection/face_detection_front.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_front.tflite"
       delegate { xnnpack {} }
     }
   }
diff --git a/mediapipe/modules/face_detection/face_detection_front_gpu.pbtxt b/mediapipe/modules/face_detection/face_detection_front_gpu.pbtxt
index 74bc46d..1e828e3 100644
--- a/mediapipe/modules/face_detection/face_detection_front_gpu.pbtxt
+++ b/mediapipe/modules/face_detection/face_detection_front_gpu.pbtxt
@@ -59,7 +59,7 @@ node {
   output_stream: "TENSORS_GPU:detection_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/modules/face_detection/face_detection_front.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_detection_front.tflite"
     }
   }
 }
diff --git a/mediapipe/modules/face_landmark/face_landmark_cpu.pbtxt b/mediapipe/modules/face_landmark/face_landmark_cpu.pbtxt
index 66ecf60..ba64a99 100644
--- a/mediapipe/modules/face_landmark/face_landmark_cpu.pbtxt
+++ b/mediapipe/modules/face_landmark/face_landmark_cpu.pbtxt
@@ -73,7 +73,7 @@ node {
   output_stream: "TENSORS:output_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/modules/face_landmark/face_landmark.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_landmark.tflite"
       delegate { xnnpack {} }
     }
   }
diff --git a/mediapipe/modules/face_landmark/face_landmark_gpu.pbtxt b/mediapipe/modules/face_landmark/face_landmark_gpu.pbtxt
index 3460d76..075e0c7 100644
--- a/mediapipe/modules/face_landmark/face_landmark_gpu.pbtxt
+++ b/mediapipe/modules/face_landmark/face_landmark_gpu.pbtxt
@@ -73,7 +73,7 @@ node {
   output_stream: "TENSORS:output_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/modules/face_landmark/face_landmark.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/face_landmark.tflite"
     }
   }
 }
diff --git a/mediapipe/modules/iris_landmark/iris_landmark_cpu.pbtxt b/mediapipe/modules/iris_landmark/iris_landmark_cpu.pbtxt
index f2c4b04..7135eaf 100644
--- a/mediapipe/modules/iris_landmark/iris_landmark_cpu.pbtxt
+++ b/mediapipe/modules/iris_landmark/iris_landmark_cpu.pbtxt
@@ -81,7 +81,7 @@ node {
   output_stream: "TENSORS:output_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/modules/iris_landmark/iris_landmark.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/iris_landmark.tflite"
       delegate { xnnpack {} }
     }
   }
diff --git a/mediapipe/modules/iris_landmark/iris_landmark_gpu.pbtxt b/mediapipe/modules/iris_landmark/iris_landmark_gpu.pbtxt
index 9fb7898..9defe95 100644
--- a/mediapipe/modules/iris_landmark/iris_landmark_gpu.pbtxt
+++ b/mediapipe/modules/iris_landmark/iris_landmark_gpu.pbtxt
@@ -88,7 +88,7 @@ node {
   output_stream: "TENSORS:output_tensors"
   options: {
     [mediapipe.TfLiteInferenceCalculatorOptions.ext] {
-      model_path: "mediapipe/modules/iris_landmark/iris_landmark.tflite"
+      model_path: "Assets/Mediapipe/SDK/Models/iris_landmark.tflite"
     }
   }
 }
