diff --git a/apple/internal/ios_rules.bzl b/apple/internal/ios_rules.bzl
index fcf84bd0..f040baa8 100644
--- a/apple/internal/ios_rules.bzl
+++ b/apple/internal/ios_rules.bzl
@@ -646,6 +646,7 @@ def _ios_framework_impl(ctx):
     apple_toolchain_info = ctx.attr._toolchain[AppleSupportToolchainInfo]
     bin_root_path = ctx.bin_dir.path
     bundle_id = ctx.attr.bundle_id
+    bundling_support.validate_bundle_id(bundle_id)
     bundle_name, bundle_extension = bundling_support.bundle_full_name_from_rule_ctx(ctx)
     executable_name = bundling_support.executable_name(ctx)
     entitlements = entitlements_support.entitlements(
