diff --git a/mediapipe/modules/objectron/calculators/BUILD b/mediapipe/modules/objectron/calculators/BUILD
index 0a8b326..a1e0e09 100644
--- a/mediapipe/modules/objectron/calculators/BUILD
+++ b/mediapipe/modules/objectron/calculators/BUILD
@@ -170,6 +170,15 @@ cc_library(
     hdrs = [
         "tensor_util.h",
     ],
+    copts = select({
+        # Needed for "//mediapipe/framework/formats:tensor" compatibility on Apple
+        # platforms for Metal pulled in via the tensor.h header.
+        "//mediapipe:apple": [
+            "-x objective-c++",
+            "-fobjc-arc",  # enable reference-counting
+        ],
+        "//conditions:default": [],
+    }),
     deps = [
         "//mediapipe/framework/formats:tensor",
         "//mediapipe/framework/port:logging",
@@ -285,6 +294,15 @@ cc_library(
     name = "lift_2d_frame_annotation_to_3d_calculator",
     srcs = ["lift_2d_frame_annotation_to_3d_calculator.cc"],
     visibility = ["//visibility:public"],
+    copts = select({
+        # Needed for "//mediapipe/framework/formats:tensor" compatibility on Apple
+        # platforms for Metal pulled in via the tensor.h header.
+        "//mediapipe:apple": [
+            "-x objective-c++",
+            "-fobjc-arc",  # enable reference-counting
+        ],
+        "//conditions:default": [],
+    }),
     deps = [
         ":annotation_cc_proto",
         ":belief_decoder_config_cc_proto",
