diff --git a/mediapipe/calculators/audio/BUILD b/mediapipe/calculators/audio/BUILD
index 4a8f0f59..2a911906 100644
--- a/mediapipe/calculators/audio/BUILD
+++ b/mediapipe/calculators/audio/BUILD
@@ -358,3 +358,9 @@ cc_test(
         "@eigen_archive//:eigen3",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/core/BUILD b/mediapipe/calculators/core/BUILD
index df54c580..18cfe311 100644
--- a/mediapipe/calculators/core/BUILD
+++ b/mediapipe/calculators/core/BUILD
@@ -1377,3 +1377,9 @@ cc_test(
         "@com_google_absl//absl/types:optional",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/image/BUILD b/mediapipe/calculators/image/BUILD
index 9aae8cfb..c83714e6 100644
--- a/mediapipe/calculators/image/BUILD
+++ b/mediapipe/calculators/image/BUILD
@@ -792,3 +792,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/tensor/BUILD b/mediapipe/calculators/tensor/BUILD
index 69d66609..20fe1c60 100644
--- a/mediapipe/calculators/tensor/BUILD
+++ b/mediapipe/calculators/tensor/BUILD
@@ -1359,3 +1359,9 @@ cc_test(
         "@com_google_absl//absl/status",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/tflite/BUILD b/mediapipe/calculators/tflite/BUILD
index db2a2763..6dd0cc79 100644
--- a/mediapipe/calculators/tflite/BUILD
+++ b/mediapipe/calculators/tflite/BUILD
@@ -551,3 +551,9 @@ cc_test(
         "@org_tensorflow//tensorflow/lite:framework",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/util/BUILD b/mediapipe/calculators/util/BUILD
index a679a80f..14cab7d3 100644
--- a/mediapipe/calculators/util/BUILD
+++ b/mediapipe/calculators/util/BUILD
@@ -1414,3 +1414,9 @@ cc_test(
         "@com_google_absl//absl/strings",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/video/BUILD b/mediapipe/calculators/video/BUILD
index f2b8135f..25c503d1 100644
--- a/mediapipe/calculators/video/BUILD
+++ b/mediapipe/calculators/video/BUILD
@@ -529,3 +529,9 @@ cc_test(
         "//mediapipe/framework/port:status",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/framework/BUILD b/mediapipe/framework/BUILD
index da8ef3b3..689257a8 100644
--- a/mediapipe/framework/BUILD
+++ b/mediapipe/framework/BUILD
@@ -1684,3 +1684,10 @@ cc_test(
         "@com_google_absl//absl/strings:str_format",
     ],
 )
+
+# Expose the proto source files for building mediapipe AAR.
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/framework/formats/BUILD b/mediapipe/framework/formats/BUILD
index 10aa3fca..68022af7 100644
--- a/mediapipe/framework/formats/BUILD
+++ b/mediapipe/framework/formats/BUILD
@@ -312,6 +312,13 @@ mediapipe_register_type(
     deps = [":landmark_cc_proto"],
 )
 
+# Expose the proto source files for building mediapipe AAR.
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "image",
     srcs = ["image.cc"],
diff --git a/mediapipe/framework/formats/annotation/BUILD b/mediapipe/framework/formats/annotation/BUILD
index 9bcb7bcc..ec795bd0 100644
--- a/mediapipe/framework/formats/annotation/BUILD
+++ b/mediapipe/framework/formats/annotation/BUILD
@@ -30,4 +30,12 @@ mediapipe_proto_library(
 mediapipe_proto_library(
     name = "rasterization_proto",
     srcs = ["rasterization.proto"],
+    visibility = ["//visibility:public"],
+)
+
+# Expose the proto source files for building mediapipe AAR.
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/framework/formats/motion/BUILD b/mediapipe/framework/formats/motion/BUILD
index c9bb8b4f..d359e394 100644
--- a/mediapipe/framework/formats/motion/BUILD
+++ b/mediapipe/framework/formats/motion/BUILD
@@ -72,3 +72,9 @@ cc_test(
         "@org_tensorflow//tensorflow/core:framework",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/framework/formats/object_detection/BUILD b/mediapipe/framework/formats/object_detection/BUILD
index 35292e1c..3071c516 100644
--- a/mediapipe/framework/formats/object_detection/BUILD
+++ b/mediapipe/framework/formats/object_detection/BUILD
@@ -31,3 +31,9 @@ mediapipe_cc_proto_library(
     srcs = ["anchor.proto"],
     deps = [":anchor_proto"],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/gpu/BUILD b/mediapipe/gpu/BUILD
index 55e9c98c..1ea5f3c6 100644
--- a/mediapipe/gpu/BUILD
+++ b/mediapipe/gpu/BUILD
@@ -1187,3 +1187,9 @@ mediapipe_cc_test(
         "//mediapipe/framework/port:gtest_main",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/graphs/instant_motion_tracking/calculators/BUILD b/mediapipe/graphs/instant_motion_tracking/calculators/BUILD
index 93af68c2..2ef17aff 100644
--- a/mediapipe/graphs/instant_motion_tracking/calculators/BUILD
+++ b/mediapipe/graphs/instant_motion_tracking/calculators/BUILD
@@ -82,3 +82,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/graphs/iris_tracking/calculators/BUILD b/mediapipe/graphs/iris_tracking/calculators/BUILD
index f5124b46..a11ef3ba 100644
--- a/mediapipe/graphs/iris_tracking/calculators/BUILD
+++ b/mediapipe/graphs/iris_tracking/calculators/BUILD
@@ -104,3 +104,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/graphs/object_detection_3d/calculators/BUILD b/mediapipe/graphs/object_detection_3d/calculators/BUILD
index d4c5c496..2e0b92f8 100644
--- a/mediapipe/graphs/object_detection_3d/calculators/BUILD
+++ b/mediapipe/graphs/object_detection_3d/calculators/BUILD
@@ -116,3 +116,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/java/com/google/mediapipe/components/BUILD b/mediapipe/java/com/google/mediapipe/components/BUILD
index a1ec1754..012ae985 100644
--- a/mediapipe/java/com/google/mediapipe/components/BUILD
+++ b/mediapipe/java/com/google/mediapipe/components/BUILD
@@ -91,5 +91,5 @@ android_library(
 filegroup(
     name = "java_src",
     srcs = glob(["*.java"]),
-    visibility = ["//mediapipe:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/java/com/google/mediapipe/framework/BUILD b/mediapipe/java/com/google/mediapipe/framework/BUILD
index dd5f8f1d..7fa8935a 100644
--- a/mediapipe/java/com/google/mediapipe/framework/BUILD
+++ b/mediapipe/java/com/google/mediapipe/framework/BUILD
@@ -145,5 +145,5 @@ filegroup(
         ["*.java"],
         exclude = ["TypeNameRegistryFull.java"],
     ),
-    visibility = ["//mediapipe:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/java/com/google/mediapipe/glutil/BUILD b/mediapipe/java/com/google/mediapipe/glutil/BUILD
index b1c89240..cd104754 100644
--- a/mediapipe/java/com/google/mediapipe/glutil/BUILD
+++ b/mediapipe/java/com/google/mediapipe/glutil/BUILD
@@ -35,5 +35,5 @@ android_library(
 filegroup(
     name = "java_src",
     srcs = glob(["**/*.java"]),
-    visibility = ["//mediapipe:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/modules/face_detection/BUILD b/mediapipe/modules/face_detection/BUILD
index 12abfbb3..20ae41d0 100644
--- a/mediapipe/modules/face_detection/BUILD
+++ b/mediapipe/modules/face_detection/BUILD
@@ -177,3 +177,9 @@ mediapipe_files(
         "face_detection_short_range.tflite",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/face_geometry/BUILD b/mediapipe/modules/face_geometry/BUILD
index c1f99675..f5836d03 100644
--- a/mediapipe/modules/face_geometry/BUILD
+++ b/mediapipe/modules/face_geometry/BUILD
@@ -135,3 +135,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/face_geometry/protos/BUILD b/mediapipe/modules/face_geometry/protos/BUILD
index 48b7b66c..2d53a28d 100644
--- a/mediapipe/modules/face_geometry/protos/BUILD
+++ b/mediapipe/modules/face_geometry/protos/BUILD
@@ -44,3 +44,9 @@ mediapipe_proto_library(
     name = "mesh_3d_proto",
     srcs = ["mesh_3d.proto"],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/holistic_landmark/calculators/BUILD b/mediapipe/modules/holistic_landmark/calculators/BUILD
index bc00b697..374b437e 100644
--- a/mediapipe/modules/holistic_landmark/calculators/BUILD
+++ b/mediapipe/modules/holistic_landmark/calculators/BUILD
@@ -58,3 +58,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/objectron/calculators/BUILD b/mediapipe/modules/objectron/calculators/BUILD
index 14cea526..bd5a2807 100644
--- a/mediapipe/modules/objectron/calculators/BUILD
+++ b/mediapipe/modules/objectron/calculators/BUILD
@@ -418,3 +418,9 @@ cc_test(
         "@com_google_absl//absl/container:flat_hash_set",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/util/BUILD b/mediapipe/util/BUILD
index 55556955..ccb1d87b 100644
--- a/mediapipe/util/BUILD
+++ b/mediapipe/util/BUILD
@@ -387,3 +387,9 @@ cc_library(
         "//mediapipe/framework/port:opencv_imgproc",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/util/tracking/BUILD b/mediapipe/util/tracking/BUILD
index 816af253..6b9f583d 100644
--- a/mediapipe/util/tracking/BUILD
+++ b/mediapipe/util/tracking/BUILD
@@ -783,3 +783,9 @@ cc_library(
         "@com_google_absl//absl/container:node_hash_map",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
