diff --git a/mediapipe/calculators/audio/BUILD b/mediapipe/calculators/audio/BUILD
index 68feca713..4dcc19d4c 100644
--- a/mediapipe/calculators/audio/BUILD
+++ b/mediapipe/calculators/audio/BUILD
@@ -503,3 +503,9 @@ cc_test(
         "@eigen//:eigen3",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/core/BUILD b/mediapipe/calculators/core/BUILD
index 05d5f48df..5ea81d1f8 100644
--- a/mediapipe/calculators/core/BUILD
+++ b/mediapipe/calculators/core/BUILD
@@ -1513,3 +1513,9 @@ cc_test(
         "//mediapipe/framework/port:gtest_main",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/image/BUILD b/mediapipe/calculators/image/BUILD
index 5e790bd75..baa03075f 100644
--- a/mediapipe/calculators/image/BUILD
+++ b/mediapipe/calculators/image/BUILD
@@ -943,3 +943,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/tensor/BUILD b/mediapipe/calculators/tensor/BUILD
index 583ef55b6..cceb618a7 100644
--- a/mediapipe/calculators/tensor/BUILD
+++ b/mediapipe/calculators/tensor/BUILD
@@ -2291,3 +2291,9 @@ cc_test(
         "@com_google_absl//absl/strings",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/tflite/BUILD b/mediapipe/calculators/tflite/BUILD
index 2526088d1..4fa72db4b 100644
--- a/mediapipe/calculators/tflite/BUILD
+++ b/mediapipe/calculators/tflite/BUILD
@@ -576,3 +576,9 @@ cc_test(
         "@org_tensorflow//tensorflow/lite/kernels:builtin_ops",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/util/BUILD b/mediapipe/calculators/util/BUILD
index 8bff2e22d..190456449 100644
--- a/mediapipe/calculators/util/BUILD
+++ b/mediapipe/calculators/util/BUILD
@@ -1893,3 +1893,9 @@ cc_test(
         "@com_google_absl//absl/status",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/calculators/video/BUILD b/mediapipe/calculators/video/BUILD
index eb833732b..d230de23a 100644
--- a/mediapipe/calculators/video/BUILD
+++ b/mediapipe/calculators/video/BUILD
@@ -476,3 +476,9 @@ cc_test(
         "//mediapipe/framework/port:status",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/framework/BUILD b/mediapipe/framework/BUILD
index 9ea70a826..e64f0fecc 100644
--- a/mediapipe/framework/BUILD
+++ b/mediapipe/framework/BUILD
@@ -1988,3 +1988,10 @@ cc_library(
         "@com_google_absl//absl/strings:string_view",
     ],
 )
+
+# Expose the proto source files for building mediapipe AAR.
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/framework/formats/BUILD b/mediapipe/framework/formats/BUILD
index d290e7ebe..61a0c38b0 100644
--- a/mediapipe/framework/formats/BUILD
+++ b/mediapipe/framework/formats/BUILD
@@ -448,6 +448,13 @@ mediapipe_register_type(
     deps = [":landmark_cc_proto"],
 )
 
+# Expose the proto source files for building mediapipe AAR.
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
+
 cc_library(
     name = "image",
     srcs = ["image.cc"],
diff --git a/mediapipe/framework/formats/annotation/BUILD b/mediapipe/framework/formats/annotation/BUILD
index cbbb28420..5431f28ec 100644
--- a/mediapipe/framework/formats/annotation/BUILD
+++ b/mediapipe/framework/formats/annotation/BUILD
@@ -31,4 +31,12 @@ mediapipe_proto_library(
 mediapipe_proto_library(
     name = "rasterization_proto",
     srcs = ["rasterization.proto"],
+    visibility = ["//visibility:public"],
+)
+
+# Expose the proto source files for building mediapipe AAR.
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/framework/formats/motion/BUILD b/mediapipe/framework/formats/motion/BUILD
index 5f39f79b3..8741eba4a 100644
--- a/mediapipe/framework/formats/motion/BUILD
+++ b/mediapipe/framework/formats/motion/BUILD
@@ -68,3 +68,9 @@ cc_test(
         "@org_tensorflow//tensorflow/core:framework",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/framework/formats/object_detection/BUILD b/mediapipe/framework/formats/object_detection/BUILD
index 703d5d774..a73795f16 100644
--- a/mediapipe/framework/formats/object_detection/BUILD
+++ b/mediapipe/framework/formats/object_detection/BUILD
@@ -25,3 +25,9 @@ mediapipe_proto_library(
     name = "anchor_proto",
     srcs = ["anchor.proto"],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/gpu/BUILD b/mediapipe/gpu/BUILD
index 739ac33b0..e81e14053 100644
--- a/mediapipe/gpu/BUILD
+++ b/mediapipe/gpu/BUILD
@@ -1500,3 +1500,9 @@ mediapipe_cc_test(
         "@com_google_absl//absl/log:absl_check",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/graphs/instant_motion_tracking/calculators/BUILD b/mediapipe/graphs/instant_motion_tracking/calculators/BUILD
index bad583c8f..7fd3021ea 100644
--- a/mediapipe/graphs/instant_motion_tracking/calculators/BUILD
+++ b/mediapipe/graphs/instant_motion_tracking/calculators/BUILD
@@ -84,3 +84,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/graphs/iris_tracking/calculators/BUILD b/mediapipe/graphs/iris_tracking/calculators/BUILD
index 9ddce7f36..a9197a9fc 100644
--- a/mediapipe/graphs/iris_tracking/calculators/BUILD
+++ b/mediapipe/graphs/iris_tracking/calculators/BUILD
@@ -84,3 +84,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/graphs/object_detection_3d/calculators/BUILD b/mediapipe/graphs/object_detection_3d/calculators/BUILD
index d89891a7b..8465f77a6 100644
--- a/mediapipe/graphs/object_detection_3d/calculators/BUILD
+++ b/mediapipe/graphs/object_detection_3d/calculators/BUILD
@@ -118,3 +118,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/java/com/google/mediapipe/components/BUILD b/mediapipe/java/com/google/mediapipe/components/BUILD
index 150d16fcc..1ff6a6538 100644
--- a/mediapipe/java/com/google/mediapipe/components/BUILD
+++ b/mediapipe/java/com/google/mediapipe/components/BUILD
@@ -96,5 +96,5 @@ android_library(
 filegroup(
     name = "java_src",
     srcs = glob(["*.java"]),
-    visibility = ["//mediapipe:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/java/com/google/mediapipe/framework/BUILD b/mediapipe/java/com/google/mediapipe/framework/BUILD
index 300aea77f..a860b4416 100644
--- a/mediapipe/java/com/google/mediapipe/framework/BUILD
+++ b/mediapipe/java/com/google/mediapipe/framework/BUILD
@@ -146,5 +146,5 @@ filegroup(
         ["*.java"],
         exclude = ["TypeNameRegistryFull.java"],
     ),
-    visibility = ["//mediapipe:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/java/com/google/mediapipe/glutil/BUILD b/mediapipe/java/com/google/mediapipe/glutil/BUILD
index b1c89240e..cd104754c 100644
--- a/mediapipe/java/com/google/mediapipe/glutil/BUILD
+++ b/mediapipe/java/com/google/mediapipe/glutil/BUILD
@@ -35,5 +35,5 @@ android_library(
 filegroup(
     name = "java_src",
     srcs = glob(["**/*.java"]),
-    visibility = ["//mediapipe:__subpackages__"],
+    visibility = ["//visibility:public"],
 )
diff --git a/mediapipe/modules/face_detection/BUILD b/mediapipe/modules/face_detection/BUILD
index 05eef34ef..14f776228 100644
--- a/mediapipe/modules/face_detection/BUILD
+++ b/mediapipe/modules/face_detection/BUILD
@@ -177,3 +177,9 @@ mediapipe_files(
         "face_detection_short_range.tflite",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/face_geometry/BUILD b/mediapipe/modules/face_geometry/BUILD
index 6cb45b285..d7832ce8c 100644
--- a/mediapipe/modules/face_geometry/BUILD
+++ b/mediapipe/modules/face_geometry/BUILD
@@ -141,3 +141,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/face_geometry/protos/BUILD b/mediapipe/modules/face_geometry/protos/BUILD
index 48b7b66ce..2d53a28d9 100644
--- a/mediapipe/modules/face_geometry/protos/BUILD
+++ b/mediapipe/modules/face_geometry/protos/BUILD
@@ -44,3 +44,9 @@ mediapipe_proto_library(
     name = "mesh_3d_proto",
     srcs = ["mesh_3d.proto"],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/holistic_landmark/calculators/BUILD b/mediapipe/modules/holistic_landmark/calculators/BUILD
index bc00b697c..374b437e1 100644
--- a/mediapipe/modules/holistic_landmark/calculators/BUILD
+++ b/mediapipe/modules/holistic_landmark/calculators/BUILD
@@ -58,3 +58,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/modules/objectron/calculators/BUILD b/mediapipe/modules/objectron/calculators/BUILD
index 1d4f559c9..3e425a2e1 100644
--- a/mediapipe/modules/objectron/calculators/BUILD
+++ b/mediapipe/modules/objectron/calculators/BUILD
@@ -430,3 +430,9 @@ cc_test(
         "@com_google_absl//absl/log:absl_check",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/c/components/containers/BUILD b/mediapipe/tasks/c/components/containers/BUILD
index 646633da6..9f3f061e0 100644
--- a/mediapipe/tasks/c/components/containers/BUILD
+++ b/mediapipe/tasks/c/components/containers/BUILD
@@ -12,7 +12,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-package(default_visibility = ["//mediapipe/tasks:internal"])
+package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])
 
diff --git a/mediapipe/tasks/cc/audio/audio_classifier/proto/BUILD b/mediapipe/tasks/cc/audio/audio_classifier/proto/BUILD
index 1b3783d51..243e1a842 100644
--- a/mediapipe/tasks/cc/audio/audio_classifier/proto/BUILD
+++ b/mediapipe/tasks/cc/audio/audio_classifier/proto/BUILD
@@ -28,3 +28,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/core/proto:base_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/components/containers/proto/BUILD b/mediapipe/tasks/cc/components/containers/proto/BUILD
index 66255aed7..8dab4bbc0 100644
--- a/mediapipe/tasks/cc/components/containers/proto/BUILD
+++ b/mediapipe/tasks/cc/components/containers/proto/BUILD
@@ -42,3 +42,9 @@ mediapipe_proto_library(
         "//mediapipe/framework/formats:rect_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/components/processors/proto/BUILD b/mediapipe/tasks/cc/components/processors/proto/BUILD
index 82d4ea21b..9b14e8ef2 100644
--- a/mediapipe/tasks/cc/components/processors/proto/BUILD
+++ b/mediapipe/tasks/cc/components/processors/proto/BUILD
@@ -93,3 +93,9 @@ mediapipe_proto_library(
         "//mediapipe/framework:calculator_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/core/BUILD b/mediapipe/tasks/cc/core/BUILD
index 9c53dcca7..98eb9f137 100644
--- a/mediapipe/tasks/cc/core/BUILD
+++ b/mediapipe/tasks/cc/core/BUILD
@@ -14,7 +14,7 @@
 
 load("@org_tensorflow//tensorflow/lite/core/shims:cc_library_with_tflite.bzl", "cc_library_with_tflite", "cc_test_with_tflite")
 
-package(default_visibility = ["//mediapipe/tasks:internal"])
+package(default_visibility = ["//visibility:public"])
 
 licenses(["notice"])
 
diff --git a/mediapipe/tasks/cc/core/proto/BUILD b/mediapipe/tasks/cc/core/proto/BUILD
index 8392849af..5eaed9437 100644
--- a/mediapipe/tasks/cc/core/proto/BUILD
+++ b/mediapipe/tasks/cc/core/proto/BUILD
@@ -67,3 +67,9 @@ mediapipe_proto_library(
         "//mediapipe/framework:calculator_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/face_detector/proto/BUILD b/mediapipe/tasks/cc/vision/face_detector/proto/BUILD
index bdfe65ee7..c795ee7ef 100644
--- a/mediapipe/tasks/cc/vision/face_detector/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/face_detector/proto/BUILD
@@ -30,3 +30,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/core/proto:base_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/face_geometry/calculators/BUILD b/mediapipe/tasks/cc/vision/face_geometry/calculators/BUILD
index 3f2833f3b..1ede1db51 100644
--- a/mediapipe/tasks/cc/vision/face_geometry/calculators/BUILD
+++ b/mediapipe/tasks/cc/vision/face_geometry/calculators/BUILD
@@ -74,3 +74,9 @@ cc_library(
     ],
     alwayslink = 1,
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/face_geometry/proto/BUILD b/mediapipe/tasks/cc/vision/face_geometry/proto/BUILD
index 14bdaea80..2f5a32c0a 100644
--- a/mediapipe/tasks/cc/vision/face_geometry/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/face_geometry/proto/BUILD
@@ -64,3 +64,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/vision/face_geometry/calculators:geometry_pipeline_calculator_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/face_landmarker/proto/BUILD b/mediapipe/tasks/cc/vision/face_landmarker/proto/BUILD
index aa839d912..8efc28e15 100644
--- a/mediapipe/tasks/cc/vision/face_landmarker/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/face_landmarker/proto/BUILD
@@ -64,3 +64,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/vision/face_geometry/proto:face_geometry_graph_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/gesture_recognizer/proto/BUILD b/mediapipe/tasks/cc/vision/gesture_recognizer/proto/BUILD
index 8e4f28060..3a67358c7 100644
--- a/mediapipe/tasks/cc/vision/gesture_recognizer/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/gesture_recognizer/proto/BUILD
@@ -64,3 +64,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/vision/hand_landmarker/proto:hand_landmarker_graph_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/hand_detector/proto/BUILD b/mediapipe/tasks/cc/vision/hand_detector/proto/BUILD
index c1453f420..2339661a0 100644
--- a/mediapipe/tasks/cc/vision/hand_detector/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/hand_detector/proto/BUILD
@@ -38,3 +38,9 @@ mediapipe_proto_library(
         "//mediapipe/framework/formats:rect_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/hand_landmarker/proto/BUILD b/mediapipe/tasks/cc/vision/hand_landmarker/proto/BUILD
index 6f79b9c4f..f65ebb86f 100644
--- a/mediapipe/tasks/cc/vision/hand_landmarker/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/hand_landmarker/proto/BUILD
@@ -52,3 +52,9 @@ mediapipe_proto_library(
 )
 
 # TODO: open source hand joints graph
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/holistic_landmarker/proto/BUILD b/mediapipe/tasks/cc/vision/holistic_landmarker/proto/BUILD
index 147f3cc86..da1d7f571 100644
--- a/mediapipe/tasks/cc/vision/holistic_landmarker/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/holistic_landmarker/proto/BUILD
@@ -42,3 +42,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/vision/pose_landmarker/proto:pose_landmarks_detector_graph_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/image_segmenter/calculators/BUILD b/mediapipe/tasks/cc/vision/image_segmenter/calculators/BUILD
index a694f9e27..349510eb8 100644
--- a/mediapipe/tasks/cc/vision/image_segmenter/calculators/BUILD
+++ b/mediapipe/tasks/cc/vision/image_segmenter/calculators/BUILD
@@ -125,3 +125,9 @@ cc_test(
         "@com_google_absl//absl/types:span",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/image_segmenter/proto/BUILD b/mediapipe/tasks/cc/vision/image_segmenter/proto/BUILD
index 54dc399d3..c1c4db012 100644
--- a/mediapipe/tasks/cc/vision/image_segmenter/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/image_segmenter/proto/BUILD
@@ -33,3 +33,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/core/proto:base_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/object_detector/proto/BUILD b/mediapipe/tasks/cc/vision/object_detector/proto/BUILD
index 863d69df0..3abbc14a7 100644
--- a/mediapipe/tasks/cc/vision/object_detector/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/object_detector/proto/BUILD
@@ -27,3 +27,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/core/proto:base_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/pose_detector/proto/BUILD b/mediapipe/tasks/cc/vision/pose_detector/proto/BUILD
index 53e7d5a55..3882cb6a6 100644
--- a/mediapipe/tasks/cc/vision/pose_detector/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/pose_detector/proto/BUILD
@@ -29,3 +29,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/core/proto:base_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/tasks/cc/vision/pose_landmarker/proto/BUILD b/mediapipe/tasks/cc/vision/pose_landmarker/proto/BUILD
index 869a1ea60..3cd6bc5af 100644
--- a/mediapipe/tasks/cc/vision/pose_landmarker/proto/BUILD
+++ b/mediapipe/tasks/cc/vision/pose_landmarker/proto/BUILD
@@ -41,3 +41,9 @@ mediapipe_proto_library(
         "//mediapipe/tasks/cc/vision/pose_detector/proto:pose_detector_graph_options_proto",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/util/BUILD b/mediapipe/util/BUILD
index cb801e88d..d2a14193e 100644
--- a/mediapipe/util/BUILD
+++ b/mediapipe/util/BUILD
@@ -542,3 +542,9 @@ cc_test(
         "@com_google_absl//absl/strings:string_view",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
diff --git a/mediapipe/util/tracking/BUILD b/mediapipe/util/tracking/BUILD
index df3f00446..906ac2600 100644
--- a/mediapipe/util/tracking/BUILD
+++ b/mediapipe/util/tracking/BUILD
@@ -686,3 +686,9 @@ cc_library(
         "@com_google_absl//absl/container:node_hash_map",
     ],
 )
+
+filegroup(
+    name = "protos_src",
+    srcs = glob(["*.proto"]),
+    visibility = ["//visibility:public"],
+)
